Shell/main.cpp

```diff
@@ -1487,6 +1714,13 @@ int main(int argc, char** argv)
         save_history();
     });
 
+    signal(SIGCHLD, [](int) {
+        auto child_pid = waitpid(-1, nullptr, WNOHANG);
+        if (child_pid != -1) {
+            g.jobs.remove(child_pid);
```

Comment:
Don't just remove it, we have to print the info ("Job foo terminated")

Reply:
printing in signal handlers just...scares me

Reply:
Don't print from the signal handler, that's indeed scary. Just stash the info away for later. SystemServer uses a deferred invocation for this, dunno if the shell can do the same (does it even use `Core::EventLoop`?), but something like that:

https://github.com/SerenityOS/serenity/blob/cd298446328af0d71a74d31f750fb1326689a3fe/Services/SystemServer/main.cpp#L55-L62

Reply:
(Although SystemServer does in fact `dbg()` from the handler, and it works just fine).

Reply:
I guess we change the shell loop to work with `Core::EventLoop`, and this stuff would be magically better

Reply:
This PR will do a very silly thing, and print from the signal handler...we should _really_ switch to Core::EventLoop asap